networkx/algorithms/community/tests/test_louvain.py

```diff
@@ -0,0 +1,115 @@
+from random import randint, seed
+
+import pytest
+
+import networkx as nx
+from networkx.algorithms.community import (
+    louvain_communities,
+    modularity,
+    is_partition,
+    partition_quality,
+)
+
+
+def test_modularity_increase():
+    G = nx.LFR_benchmark_graph(
+        250, 3, 1.5, 0.009, average_degree=5, min_community=20, seed=10
+    )
+    partition = [{u} for u in G.nodes()]
+    mod = modularity(G, partition)
+    partition = louvain_communities(G)
+
+    assert modularity(G, partition) > mod
+
+
+def test_valid_partition():
+    G = nx.LFR_benchmark_graph(
+        250, 3, 1.5, 0.009, average_degree=5, min_community=20, seed=10
+    )
+    H = G.to_directed()
+    partition = louvain_communities(G)
+    partition2 = louvain_communities(H)
+
+    assert is_partition(G, partition)
+    assert is_partition(H, partition2)
+
+
+def test_partition():
+    G = nx.karate_club_graph()
+    part = [
+        {0, 1, 2, 3, 7, 9, 11, 12, 13, 17, 19, 21},
+        {16, 4, 5, 6, 10},
+        {23, 25, 27, 28, 24, 31},
+        {32, 33, 8, 14, 15, 18, 20, 22, 26, 29, 30},
+    ]
+    partition = louvain_communities(G, seed=2)
+
+    assert part == partition
+
+
+def test_none_weight_param():
+    G = nx.karate_club_graph()
+    seed(1234)
+    nx.set_edge_attributes(G, {edge: randint(1, 20) for edge in G.edges}, name="foo")
```

Comment:
This might be a better test by not using `randint`, but instead use `enumerate` or some non-random way to generate the weights. That eliminates any possible intermittent test failures due to random results (which shouldn't happen here, but it's good to eliminate that source of debugging effort to make sure it isn't the part causing the errors).    With this and seed removed you can remove the `import` statement at the top of the module.